### PR TITLE
removing unnecessary removeLayer call, prevents folderId access w/in …

### PR DESCRIPTION
**Pull request by jackgrossman18**
ref #1754

### CODE REVIEW

File: modules/Hoot/ui/conflicts/resolve.js [hunk 1]
```diff
@@ -3,8 +3,10 @@
  * Project: hootenanny-ui
  * @author Matt Putipong - matt.putipong@radiantsolutions.com on 5/24/18
  *******************************************************************************************************/
+import _forEach from 'lodash-es/forEach';
 
 import PublishBookmark from '../modals/publishBookmark';
+import { load } from 'js-yaml';
```
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/resolve.js]: is this used?

File: modules/Hoot/ui/conflicts/resolve.js [hunk 2]
```diff
@@ -82,20 +84,20 @@ export default class Resolve {
...
+        let conflateController = this.sidebar.forms.conflate.controller;
+
+        let loadedLayers = Hoot.layers.loadedLayers;
+
+        _forEach( loadedLayers, lyr => {
+
+            if ( !lyr.isMerged ) {
+                Hoot.layers.removeLoadedLayer( lyr.id );
+            }
+        } );
```
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/resolve.js]: This fixes the bug, but introduces another.  By removing the source layers, the "show inputs" button is broken.
image

This makes me wonder what the point of removing the merged layer, only to add it back in might be.  Maybe we don't need to do that at all.

Maybe just forcing a reload of the layer data with review relations updated to resolved?  Would have to look at the http request that Resolve All Remaining makes, it might update the review relations on the back end, then maybe there is another method in iD to force a reload of the tiled layer data.

**Review by brianhatchl — CHANGES_REQUESTED** [review]

**Comment by jackgrossman18** [comment]
@brianhatchl I added a method to the sidebar controller class to hide the switch between the inputs of the merged layer, would this be suitable instead of forcing a reload of the layer data?

**Comment by brianhatchl** [comment]
I think we want to keep it because it is still a merged layer and should act like if we added the merged layer from the add datasource button.

Maybe just try removing these two lines from the original:
```
        // update layer	
        Hoot.layers.removeLayer( layer.id );	
        Hoot.layers.loadLayer( key, true );	
```

I noticed other defects like when I go to resolve all remaining the count shown is zero when there are unresolved reviews.  Have you noticed this?

**Comment by jackgrossman18** [comment]
@brianhatchl I added the proper count for the unresolved reviews, also updated clipDataset

File: modules/Hoot/ui/sidebar/sidebarController.js [hunk 3]
```diff
@@ -182,6 +182,10 @@ class SidebarController {
             .text('delete_outline');
     }
 
+    hideShowLayersButton() {
+        d3.select('.showlayers').classed('hidden', true);
+    }
+
```
Comment (brianhatchl) [inline modules/Hoot/ui/sidebar/sidebarController.js]: no longer needed

File: modules/Hoot/ui/conflicts/resolve.js [hunk 4]
```diff
@@ -82,20 +82,11 @@ export default class Resolve {
...
 
         // enter controller refresh state
         conflateController.text.html( 'Refreshing &#8230;' );
 
-        // update layer
-        Hoot.layers.removeLayer( layer.id );
-        Hoot.layers.loadLayer( key, true );
-
         // exit controller refresh state
         conflateController.text.html( layer.name );
```
Comment (brianhatchl) [inline modules/Hoot/ui/conflicts/resolve.js]: probably don't need these anymore

File: modules/Hoot/tools/clipDataset.js [hunk 5]
```diff
@@ -93,7 +94,17 @@ export default class ClipDataset {
...
+            if (layer.isMerged) {
+                return Hoot.layers.loadedLayers[ layer.id ] = layer;
+            }
+        } );
+
         _forEach( loadedLayers, layer => {
+            if ( checkMerge && checkMerge.id !== layer.id  ) {
+                return;
+            }
+
```
Comment (brianhatchl) [inline modules/Hoot/tools/clipDataset.js]: I find this kinda hard to follow.  Maybe something more "fluent" like:
```
        let loadedList = Object.values(loadedLayers);
        let clipLayers = loadedList.some(l => l.isMerged) ? loadedList.filter(l => l.isMerged) : loadedList;
        _forEach( clipLayers, layer => {

```

**Review by brianhatchl — CHANGES_REQUESTED** [review]

**Review by brianhatchl — APPROVED** [review]
